src/modules/linux-micro/machine-id/machine-id.c

```diff
@@ -94,7 +94,7 @@ run_do(void)
         SOL_INT_CHECK(r, == 0, 0);
     }
 
-    r = sol_util_uuid_gen(false, true, id);
+    r = sol_util_uuid_gen(false, false, id);
```

Comment:
Might be a good idea to have this function take an `enum sol_util_uuid_flags { SOL_UTIL_UUID_UPPER_CASE = 1<<0, SOL_UTIL_UUID_WITH_HYPHENS = 1<<1 }`.